test/http.js

```diff
@@ -115,6 +133,31 @@ test('response contains url', async t => {
 	t.is((await got(s.url)).url, `${s.url}/`);
 });
 
+test.serial('cancel in-progress request', async t => {
+	const body = new Readable({
+		read() {}
+	});
+	body.push(String(1));
+
+	const p = got(`${s.url}/no-res-store-id`, {body});
+
+	setImmediate(() => {
+		p.cancel();
+		body.push(null); // Complete the request that should now be canceled
+	});
+
+	await t.throws(p, PCancelable.CancelError);
+	await t.notThrows(() => cancelPromise);
```

Comment:
@AlexTes `t.notThrows` either accepts a promise that rejects or a function that throws. AVA is actually catching a bug in your test here as `cancelPromise` is `null`, not a Promise. Try: `console.log(cancelPromise === null);`